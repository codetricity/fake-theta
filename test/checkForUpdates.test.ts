import {
  request,
  toMatchInvalidHeaderParameterError,
  toMatchMissingParameterError,
} from './helper';

describe('POST /osc/checkForUpdates', () => {
  it('should respond with state', async () => {
    const res = await request()
      .post('/osc/checkForUpdates')
      .send({
        stateFingerprint: 'FIG_0001',
      })
      .expect(200);
    expect(res.text).toMatchSnapshot();
  });

  it('should respond missingParameterError', async () => {
    const res = await request()
      .post('/osc/checkForUpdates')
      .send({
        fileUri: 'dummy',
      })
      .expect(400);
    toMatchMissingParameterError(res);
  });
});

describe('THETA X POST /osc/checkForUpdates', () => {
  it('should respond with state', async () => {
    const res = await request()
      .post('/osc/checkForUpdates')
      .send({
        stateFingerprint: 'FIG_0001',
      })
      .set('emulating-theta-model', 'x')
      .expect(200);
    expect(res.text).toMatchSnapshot();
  });

  it('should respond missingParameterError', async () => {
    const res = await request()
      .post('/osc/checkForUpdates')
      .send({
        fileUri: 'dummy',
      })
      .set('emulating-theta-model', 'x')
      .expect(400);
    toMatchMissingParameterError(res);
  });
});

describe('THETA Z1 POST /osc/checkForUpdates', () => {
  it('should respond with state', async () => {
    const res = await request()
      .post('/osc/checkForUpdates')
      .send({
        stateFingerprint: 'FIG_0001',
      })
      .set('emulating-theta-model', 'z1')
      .expect(200);
    expect(res.text).toMatchSnapshot();
  });

  it('should respond missingParameterError', async () => {
    const res = await request()
      .post('/osc/checkForUpdates')
      .send({
        fileUri: 'dummy',
      })
      .set('emulating-theta-model', 'x')
      .expect(400);
    toMatchMissingParameterError(res);
  });
});

describe('Unknown Type POST /osc/checkForUpdates', () => {
  it('should respond invalidHeaderParameterError', async () => {
    const res = await request()
      .post('/osc/checkForUpdates')
      .send({
        stateFingerprint: 'FIG_0001',
      })
      .set('emulating-theta-model', 'unknown')
      .expect(400);
    toMatchInvalidHeaderParameterError(res);
  });
});
