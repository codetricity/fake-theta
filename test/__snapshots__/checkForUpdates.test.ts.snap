// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /osc/checkForUpdates should respond with state 1`] = `"{"stateFingerprint":"FIG_0001","throttleTimeout":1}"`;

exports[`THETA X POST /osc/checkForUpdates should respond with state 1`] = `"{"stateFingerprint":"FIG_0001","throttleTimeout":1}"`;

exports[`THETA Z1 POST /osc/checkForUpdates should respond with state 1`] = `"{"stateFingerprint":"FIG_0001","throttleTimeout":1}"`;
